EIPS/eip-6160.md

```diff
@@ -0,0 +1,80 @@
+
+Smart contracts implementing the EIP-6160 standard MUST implement all of the functions in the `EIP-6160` interface.
+
+Smart contracts implementing the EIP-6160 standard MUST implement the `EIP-165 supportsInterface` function and MUST return the constant value `true` if *`0x01cbdea7`* is passed through the `interfaceID` argument.
+
+1. Any contract complying with [EIP-20](./eip-20.md) when extended with this EIP, **MUST** implement the following interface:
+
+```solidity
+// The EIP-165 identifier of this interface is 0xbbb8b47e
+interface IERC6160Ext20 is IERC5679Ext20, IERC_ACL_CORE {}
+```
+
+2. Any contract complying with [EIP-721](./eip-721.md) when extended with this EIP, **MUST** implement the following interface:
+
+```solidity
+// The EIP-165 identifier of this interface is 0xa75a5a72
+*interface IERC6160Ext721 is IERC5679Ext721, IERC_ACL_CORE {}
+```
+
+3. Any contract complying with [EIP-1155](./eip-1155.md) when extended with this EIP, **MUST** implement the following interface:
+
+```solidity
+// The EIP-165 identifier of this interface is 0x9f77104c
+interface IERC6160Ext1155 is IERC5679Ext1155, IERC_ACL_CORE {}
+```
+
+4. It is RECOMMENDED for compliant contracts to implement the optional extension `IERC_ACL_GENERAL`.
+5. Compliant contracts MAY implement the optional extension `IERC_ACL_METADATA`.
+
+## Rationale
+
+1. We have chosen the [EIP-5679](./eip-5679.md) standard to provide the mint & burn interface.
+2. We have chosen [EIP-5982](./eip-5982.md) standard to provide the access control interface for trustless bridge contracts to check their permissions to mint & burn `ERC20` tokens.
+3. We have chosen NOT to create new events but to require the usage of existing transfer event as required by EIP-20 EIP-721 and EIP-1155 for maximum compatibility.
+
+## Implementation
+We have made an example implementation for this EIP at [polytope-labs implementation](https://github.com/polytope-labs/multichain-native-tokens/tree/main/src/tokens)
+
+## Backwards Compatibility
+
+This EIP is designed to be compatible with EIP-20, EIP-721, EIP-1155.
+
+## Security Considerations
+
+This EIP depends on the security and soundness of the underlying bridge contract. This EIP strongly recommends that token issuers permit only trustless, audited bridge contracts with permissions to mint & burn tokens or risk infinte token minting attacks.
+
+## References
+
+¹ Arbitrum: [ICustomToken](https://github.com/OffchainLabs/token-bridge-contracts/blob/main/contracts/tokenbridge/ethereum/ICustomToken.sol),  [IArbToken](https://github.com/OffchainLabs/token-bridge-contracts/blob/main/contracts/tokenbridge/arbitrum/IArbToken.sol). Optimism: [IL2StandardERC20](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts/contracts/standards/IL2StandardERC20.sol).
```

Comment:
```suggestion
```

Please remove this whole section. We don't allow external links except as mentioned in EIP-1.